internal/workflow_deadlock_test.go

```diff
@@ -73,15 +71,17 @@ func TestDataConverterWithoutDeadlockDetection(t *testing.T) {
 		env.RegisterActivity(activityFn)
 		env.ExecuteWorkflow(workflowFn)
 		require.True(t, env.IsWorkflowCompleted())
+		return env.GetWorkflowError()
 	}
 
-	// Run with a slow converter and confirm it panics
+	// Run with a slow converter and confirm a deadlock is detected
+	conv := converter.GetDefaultDataConverter()
```

Comment:
Why was this change made, was there a change in behaviour?

Reply:
The dispatcher doesn't panic anymore on a deadlock timeout, it just returns a `workflowPanicError` similar to what a panic inside the coroutine does (if you meant the comment).

Wrt. passing conv as a parameter instead of by capture, I just felt like it is cleaner that way. This wasn't possible before since `require.Panics` mandated a specific function signature. If you'd prefer, I can revert that change.

Reply:
Yes that's what I meant, that makes sense.